pallets/nbv-storage/src/lib.rs

```diff
@@ -0,0 +1,273 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	use frame_support::{
+		BoundedVec,
+		transactional,
+		sp_io::hashing::blake2_256,
+	};
+	use scale_info::prelude::boxed::Box;
+
+	//use scale_info::TypeInfo;
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_identity::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		#[pallet::constant]
+		type XPubLen: Get<u32>;
+		#[pallet::constant]
+		type PSBTMaxLen: Get<u32>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+
+		/// Xpub and hash stored. Linked to an account identity
+		XPubStored([u8 ; 32], T::AccountId),
+		/// Removed Xpub previously linked to the account
+		XpubRemoved(T::AccountId),
+		/// The PBST was succesuflly inserted and linked to the account
+		PSBTStored(T::AccountId),
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Work in progress!
+		NotYetImplemented,
+		/// Xpub shouldn't be empty. Use the identity pallet if needed.
+		NoneValue,
+		// The xpub has already been uploaded and taken by an account
+		XPubAlreadyTaken,
+		/// The identity doesn't have an xpub
+		XPubNotFound,
+		/// The generated Hashes aren't the same
+		HashingError,
+		/// Found Invalid name on an additional field 
+		InvalidAdditionalField,
+	}
+
+	/// Stores hash-xpub pairs
+	#[pallet::storage]
+	#[pallet::getter(fn xpubs)]
+	pub(super) type Xpubs<T: Config> = StorageMap<
+		_,
+		Identity,
+		[u8 ; 32],
+		BoundedVec<u8, T::XPubLen>,
+		OptionQuery,
+	>;
+
+	/// Keeps track of what accounts own what xpub.
+	#[pallet::storage]
+	#[pallet::getter(fn xpubs_by_owner)]
+	pub(super) type XpubsByOwner<T: Config> = StorageMap<
+		_,
+		Blake2_256,
+		T::AccountId,
+		[u8 ; 32],
+		OptionQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn psbts)]
+	pub(super) type PSBTs<T: Config> = StorageMap<
+		_,
+		Blake2_256,
+		T::AccountId,
+		BoundedVec<u8, T::PSBTMaxLen>,
```

Comment:
Does Bitcoin Core (or BlueWallet) have a maximum PSBT size? @sebastianmontero

Reply:
I don't think so, it depends on the number of inputs and outputs, I guess we could set a limit?

Reply:
@amatsonkali , @3yekn shouldn't we have a relationship between the psbts and a specific proposal or trx?

Reply:
Its a convention within substrate to have an upper limit for all byte-like data that must be stored on-chain.The length limit for each PSBT is currently set at 2048 bytes on the current runtime config.
Got it, I'll add the relationship requirement on the issue #20